Sources/TokamakGTK/Shapes/Shape.swift

```diff
@@ -0,0 +1,153 @@
+  var current: CGPoint = .zero
+  var start: CGPoint = .zero
+  for element in elements {
+    switch element {
+    case let .move(to: p):
+      cairo_move_to(cr, p.x, p.y)
+      current = p
+      start = p
+
+    case let .line(to: p):
+      cairo_line_to(cr, p.x, p.y)
+      current = p
+
+    case .closeSubpath:
+      cairo_close_path(cr)
+      current = start
+
+    case let .curve(to: p, control1: c1, control2: c2):
+      cairo_curve_to(cr, c1.x, c1.y, c2.x, c2.y, p.x, p.y)
+      current = p
+
+    case let .quadCurve(to: p, control: c):
+      let c1 = CGPoint(x: (current.x + 2 * c.x) / 3, y: (current.y + 2 * c.y) / 3)
+      let c2 = CGPoint(x: (p.x + 2 * c.x) / 3, y: (p.y + 2 * c.y) / 3)
+      cairo_curve_to(cr, c1.x, c1.y, c2.x, c2.y, p.x, p.y)
+      current = p
+    }
+  }
+}
+
+extension _ShapeView: ViewDeferredToRenderer {
+  public var deferredBody: AnyView {
+    return AnyView(WidgetView(build: { _ in
+      let w = gtk_drawing_area_new()
+      bindAction(to: w!)
+      return w!
+    }) {
+    })
+  }
+
+  func bindAction(to drawingArea: UnsafeMutablePointer<GtkWidget>) {
+    drawingArea.connect(signal: "draw", closure: { widget, cr in
+      cairo_save(cr)
+
+      let width = gtk_widget_get_allocated_width(widget)
+      let height = gtk_widget_get_allocated_height(widget)
+
+      let c = (style as? Color) ?? foregroundColor ?? Color.black
+
+      // XXX TODO, don't know why my environment doesn't work.
+      var environment = EnvironmentValues()
+      environment[_ColorSchemeKey] = .light
+
+      let rgba = _ColorProxy(c).resolve(in: environment)
+      var color = GdkRGBA(red: Double(rgba.red),
+                          green: Double(rgba.green),
+                          blue: Double(rgba.blue),
+                          alpha: Double(rgba.opacity))
+
+      gdk_cairo_set_source_rgba(cr, &color)
+
+      let path = shape.path(in: CGRect(origin: .zero,
+                                       size: CGSize(width: Double(width),
+                                                    height: Double(height))))
+      let elements: [Path.Element]
+      let stroke: Bool
+      if case let .stroked(strokedPath) = path.storage {
+        elements = strokedPath.path.elements
+        stroke = true
+        let style = strokedPath.style
+
+        cairo_set_line_width(cr, style.lineWidth)
+        cairo_set_line_join(cr, style.lineJoin.cairo)
+        cairo_set_line_cap(cr, style.lineCap.cairo)
+        cairo_set_miter_limit(cr, style.miterLimit)
+        cairo_set_dash(cr, style.dash, Int32(style.dash.count), style.dashPhase)
+
+      } else {
+        elements = path.elements
+        stroke = false
+      }
+
+      if fillStyle.isEOFilled {
+        cairo_set_fill_rule(cr, cairo_fill_rule_t(rawValue: 1) /* CAIRO_FILL_RULE_EVEN_ODD */)
+      } else {
+        // This is already the default
+//        cairo_set_fill_rule(cr, cairo_fill_rule_t(rawValue: 0) /* CAIRO_FILL_RULE_WINDING */)
```

Comment:
Do we want to keep this commented?

Reply:
No, I should make it like the enum options for the strike styles. That reads quite a bit better. I'll update it.